aws/resource_aws_servicecatalog_portfolio_principal_association.go

```diff
@@ -0,0 +1,165 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
+	"strings"
+	"time"
+)
+
+func resourceAwsServiceCatalogPortfolioPrincipalAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceCatalogPortfolioPrincipalAssociationCreate,
+		Read:   resourceAwsServiceCatalogPortfolioPrincipalAssociationRead,
+		Update: resourceAwsServiceCatalogPortfolioPrincipalAssociationUpdate,
+		Delete: resourceAwsServiceCatalogPortfolioPrincipalAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"portfolio_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"principal_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsServiceCatalogPortfolioPrincipalAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	_, portfolioId, principalArn := resourceAwsServiceCatalogPortfolioPrincipalAssociationRequiredParameters(d)
+	input := servicecatalog.AssociatePrincipalWithPortfolioInput{
+		PortfolioId:   aws.String(portfolioId),
+		PrincipalARN:  aws.String(principalArn),
+		PrincipalType: aws.String("IAM"),
+	}
+	conn := meta.(*AWSClient).scconn
+	_, err := conn.AssociatePrincipalWithPortfolio(&input)
+	if err != nil {
+		return fmt.Errorf("creating Service Catalog Principal(%s)/Portfolio(%s) Association failed: %s",
+			principalArn, portfolioId, err.Error())
+	}
+
+	stateConf := resource.StateChangeConf{
+		Pending:      []string{servicecatalog.StatusCreating},
+		Target:       []string{servicecatalog.StatusAvailable},
+		Timeout:      1 * time.Minute,
+		PollInterval: 3 * time.Second,
+		Refresh: func() (interface{}, string, error) {
+			err := resourceAwsServiceCatalogPortfolioPrincipalAssociationRead(d, meta)
+			if err != nil {
+				return 42, "", err
+			}
+			if d.Id() != "" {
+				return 42, servicecatalog.StatusAvailable, err
+			}
+			return 0, servicecatalog.StatusCreating, err
+		},
+	}
+	_, err = stateConf.WaitForState()
+	return err
+}
+
+func resourceAwsServiceCatalogPortfolioPrincipalAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	id, portfolioId, principalArn := resourceAwsServiceCatalogPortfolioPrincipalAssociationRequiredParameters(d)
+	input := servicecatalog.ListPrincipalsForPortfolioInput{
+		PortfolioId: aws.String(portfolioId),
+	}
+	conn := meta.(*AWSClient).scconn
+	var pageToken = ""
+	isFound := false
+	for {
+		pageOfDetails, nextPageToken, err := resourceAwsServiceCatalogPortfolioPrincipalAssociationListPrincipalsForPortfolioPage(conn, input, &pageToken)
+		if err != nil {
+			return err
+		}
+		for _, principal := range pageOfDetails {
+			if *principal.PrincipalARN == principalArn {
```

Comment:
wrap `principal.PrincipalARN` with `aws.StringValue()`

Reply:
done (check-in to come)